gammapy/datasets/simulate.py

```diff
@@ -574,3 +574,146 @@ def run(self, dataset, observation=None):
         geom = dataset._geom
         selection = geom.contains(events.map_coord(geom))
         return events.select_row_subset(selection)
+
+
+class SimulatedObservationMaker(MapDatasetEventSampler):
+    """
+    Sample event lists for a given observation and signal models.
+
+    Signal events are sampled from the given sky models and then
+    folded with the instrument response functions.
+    To improve performance, IRFs are evaluated on a pre-defined binning,
+    not at each individual event energy / coordinate.
+
+    Parameters
+    ----------
+    energy_axis_true : `~gammapy.maps.MapAxis`
+        True energy axis.
+    energy_axis : `~gammapy.maps.MapAxis`
+        Reconstructed energy axis.
+    spatial_width : float or `~astropy.units.Quantity`
+        Spatial window size. A float is assumed to be in degree.
+    spatial_spatial_bin_size : float or `~astropy.units.Quantity`, optional
+        Pixel size used for sampling. A float is assumed to be in degree.
+        Default is 0.01 degree.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}, optional
+        Defines random number generator initialisation via the `~gammapy.utils.random.get_random_state` function.
+    oversample_energy_factor : int, optional
+        Defines an oversampling factor for the energies; it is used only when sampling
+        an energy-dependent time-varying source.
+    t_delta : `~astropy.units.Quantity`, optional
+        Time interval used to sample the time-dependent source.
```

Comment:
```suggestion
    t_delta : `~astropy.units.Quantity`, optional
        Time interval used to sample the time-dependent source. Default is 0.5 s.
```

Reply:
Note that for consistency with other arguments one could also allow for `float` assuming seconds.